src/modules/user/model/index.js

```diff
@@ -1 +1,44 @@
-// hotel model here
+const { DataTypes } = require('sequelize');
+
+const sequelize = require('../../../config/SequelizeConfig');
+
+const EGender = {
+  MALE: 'MALE',
+  FEMALE: 'FEMALE',
+};
+
+const User = sequelize.define('users', {
+  name: DataTypes.STRING,
+  email: DataTypes.STRING,
+  gender: DataTypes.ENUM(Object.keys(EGender)),
+  username: DataTypes.STRING,
+  password: DataTypes.STRING,
+  birthdate: DataTypes.DATE,
+  preferred_language: DataTypes.STRING,
+  preferred_currency: DataTypes.STRING,
+  address: DataTypes.STRING,
+  role: DataTypes.STRING,
+  department: DataTypes.STRING,
+  line_manager: DataTypes.STRING
+}, {
+  timestamps: true,
+  createdAt: 'created_at',
+  updatedAt: 'updated_at'
+});
+
+User.hasOne(User, {
+  foreignKey: 'line_manager'
+});
+
+const registrationSchema = {
+  name: ['required', 'string'],
+  gender: ['required', 'in:MALE,FEMALE'],
+  email: ['required', 'string', 'email'],
+  username: ['required', 'min:3'],
+  password: ['required', 'string', 'alpha_num', 'min:8'],
+};
+
```

Comment:
- [ ] I suggest that you use `first name`, and `last name` instead of `name` attributes and make sure to validate a name may be by `length` etc currently even if I send a number as a name request is being sent

- [ ] Validate a password to be strong

Reply:
Working on that.

For the password 
what rules can we use 
like
- include at least one capital letter,
- include at least one small letter,
- include at least one symbol.
- include at least one number
- a minimum of 8
- and a maximum of 16.

any other suggestion?

Reply:
@irumvanselme yes actually use that